src/bio/bio_xstream.c

```diff
@@ -1716,6 +1719,41 @@ scan_bio_bdevs(struct bio_xs_context *ctxt, uint64_t now)
 	nvme_glb.bd_scan_age = now;
 }
 
+void
+bio_led_event_monitor(struct bio_xs_context *ctxt, uint64_t now)
+{
+	struct bio_bdev         *d_bdev;
+	struct spdk_bdev        *bdev;
+	static uint64_t          led_event_period = NVME_LED_EVENT_PERIOD;
+
+	for (bdev = spdk_bdev_first(); bdev != NULL;
+	     bdev = spdk_bdev_next(bdev)) {
+		d_bdev = lookup_dev_by_name(spdk_bdev_get_name(bdev));
+		if (d_bdev == NULL)
+			continue;
+
+		/* Transport addr only gets alloc'd on an LED event command */
+		if (d_bdev->bb_led_traddr != NULL) {
+			if (strlen(d_bdev->bb_led_traddr) == 0)
+				continue;
+			/* Init the start time of the LED event */
+			if (d_bdev->bb_led_start_time == 0) {
+				d_bdev->bb_led_start_time = now;
+				return;
+			}
+			/*
+			 * TODO: Make NVME_LED_EVENT_PERIOD configurable from
+			 * command line
+			 */
+			if (d_bdev->bb_led_start_time + led_event_period >= now)
+				return;
+
+			if (bio_set_led_state_orig(ctxt, d_bdev) != 0)
```

Comment:
re-org the two functions to avoid dup code. (bio_set_led_state_orig() and bio_set_led_state())